staging/src/k8s.io/component-base/traces/utils.go

```diff
@@ -26,33 +26,64 @@ import (
 	"go.opentelemetry.io/otel/propagation"
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/semconv"
 	"go.opentelemetry.io/otel/trace"
 
 	"k8s.io/client-go/transport"
-	"k8s.io/klog/v2"
 )
 
-// NewProvider initializes tracing in the component, and enforces recommended tracing behavior.
-func NewProvider(ctx context.Context, baseSampler sdktrace.Sampler, resourceOpts []resource.Option, opts ...otlpgrpc.Option) trace.TracerProvider {
-	opts = append(opts, otlpgrpc.WithInsecure())
+// InitTracing initialized tracing in the component, and enforces recommended tracing behavior
+func InitTracing(ctx context.Context,
+	traceID string,
+	serviceName string,
+	samplingRate *int32,
```

Comment:
Could this take a componentbase.TracingConfiguration?  That would allow extending the config without modifying the function signature in the future.